src/extensions/firetrack/FireTrackView/FireTrackWidget.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FunctionComponent, useEffect, useReducer, useState } from 'react'
+import CanvasWidget from "../../common/CanvasWidget"
+import { useLayer, useLayers } from "../../common/CanvasWidget/CanvasWidgetLayer"
+import { createElectrodeGeometryLayer, ElectrodeLayerProps } from "../../electrodegeometry/ElectrodeGeometryWidget/electrodeGeometryLayer"
+import { SortingSelection } from "../../extensionInterface"
+import { TimeseriesData } from "../../timeseries/TimeseriesViewNew/useTimeseriesModel"
+
+const updateIndexReducer = (state: number, action: {type: 'increment'}) => {
+    return state + 1
+}
+
+
+// this is duplicate code
+const zipElectrodes = (locations: number[][], ids: number[]) => {
+    if (locations && ids && ids.length !== locations.length) throw Error('Electrode ID count does not match location count.')
+    return ids.map((x, index) => {
+        const loc = locations[index]
+        return { label: x + '', x: loc[0], y: loc[1], electrodeId: x }
+    })
+}
+
+const valToColor = (v: number) => {
+    if (v <= 0) return 'black'
+    if (v >= 1) return 'white'
+    const x = Math.floor(v * 255)
+    return `rgb(${x}, ${x}, ${x})`
+}
```

Comment:
Should this live with the `CanvasPainter` code that handles color strings?

Reply:
We need to create colormaps. A color map takes a value between 0 and 1 and returns a color. I wonder if we can grab the colormaps from matplotlib somehow: https://matplotlib.org/3.3.3/tutorials/colors/colormaps.html